0000-catalog.md

```diff
@@ -0,0 +1,801 @@
+information can be found [here](https://sawtooth.hyperledger.org/docs/core/releases/1.0/transaction_family_specifications/settings_transaction_family.html).
+
+Validation requirements:
+
+- If a catalog_product with catalog_product_id does not exist, the transaction is
+  invalid.
+- The signer of the transaction must be an agent in the Pike state and must
+  belong to an organization in Pike state, otherwise the transaction is invalid.
+- The owner in the product must match the organization that the agent belongs
+  to, otherwise the transaction is invalid.
+- The agent must have the permission “can_delete_product” for the organization
+  otherwise the transaction is invalid.
+
+The inputs for CatalogProductDeleteAction must include:
+
+- Address of the Agent submitting the transaction
+- Address of the Organization the Product is being deleted for
+- Address of the catalog_product to be deleted
+
+The outputs for CatalogProductDeleteAction must include:
+
+- Address of the catalog_product to be deleted
+
+**_NOTE: Deleting a catalog_product is potentially dangerous operation that
+could leave dangling references and should be done with care._**
+
+## Catalog_Product Operations
+
+### CatalogProductSetStatus
+
+CatalogProductSetStatusAction updates an existing catalog_product in state. The
+transaction should be submitted by an agent, identified by its signing key, 
+acting on behalf of an organization that corresponds to the owner in the 
+product being updated. (Organizations and agents are defined by the Pike smart 
+contract.)
+
+The implementation of the CatalogProductSetStatusAction will handle iterating 
+through the list of catalog_ids, and update the status of the 
+catalog_product(s) accordingly. Having the address of an individual 
+catalog_product be a composite key containing the catalog_id and product_id 
+enables us to easily update the status across one, all, or specific catalogs.
+
+CatalogProductSetStatusAction does not set the status for multiple 
+catalog_products. It will only change the status of a single 
+catalog_product. The nuance being that this status change can be reflected in 
+as many, or as few catalogs an entity desires.
+
+This flexibility is to enable support for use cases like:
+
+- I am discontinuing a catalog_product for entity A, B, C, but not for retailer D.
+- I want to discontinue catalog_products for all entity A, B, C, and D.
+- I am marking a catalog_product as active for entity A and B, but not C or D just yet.
+- I am making an active catalog_product inactive for entity A and B, but not C or D.
+
+```
+message CatalogProductSetStatusAction {
+    enum Status {
+        INACTIVE = 0;
+        ACTIVE = 1;
+        DISCONTINUED = 2;
+    }
+    // catalog_id and product_id are used in deriving the state address
+    repeated string catalog_ids = 1;
+    string catalog_product_id = 2;
+    Status catalog_product_status  = 4;
+    // Reason for the change
+    string reason = 5;
```

Comment:
What happens to this reason during execution? I don't see a place where it gets stored in state.

Reply:
Oops added that field after writing up the schema. 
The `reason` field would need to be defined in the catalog_product schema like so:

```
        PropertyDefinition(
            name="status_change_reason",
            data_type=PropertyDefinition.DataType.STRING,
            description="The reason for the status change",
            required=True
        ),
```

An implementation in catalog_product would look like:
```
        PropertyValue(
            name="status_change_reason",
            data_type=PropertyDefinition.DataType.STRING,
            string_value=""
        ),
```

`reason` would probably be best to keep as an empty string upon initialization until the status of the catalog_product is actually changed.


Also, I've updated that field name to be synonymous with the catalog_product schema change above.
```
    string catalog_product_id = 2;
    Status catalog_product_status  = 4;
    // Reason for the change
    `string status_change_reason = 5;`
```